File: problem_sets/week_01/cash/cash.c
#include <cs50.h>
#include <stdio.h>

// quarters (25¢), dimes (10¢), nickels (5¢), and pennies (1¢).
int calculate_cash(int cents);

int main(void)
{
    // prompt the user for change owed, in cents
    int cents;
    do
    {
        cents = get_int("Change owed: ");
    }
    while (cents < 1);

    int coins = calculate_cash(cents);
    // Print that sum
    printf("%i\n", coins);
}

// calculate total coins
int calculate_cash(int cents)
{
    int quarters = 0;
    int dimes = 0;
    int nickels = 0;
    int pennies = 0;

    // greed algorithms
    while (cents > 0)
    {
        // Calculate how many quarters you should give customer
        if (cents >= 25)
        {
            quarters++;
            // Subtract the value of those quarters from cents
            cents = cents - 25;
        }
        // calculate how many dimes you should give customer.
        else if (cents >= 10)
        {
            dimes++;
            // Subtract the value of those quarters from cents
            cents = cents - 10;
        }
        // Calculate how many nickels you should give customer.
        else if (cents >= 5)
        {
            nickels++;
            // Subtract the value of those nickels from remaining cents
            cents = cents - 5;
        }
        else
        {
            // calculate how many pennies you should give customer.
            pennies++;
            // Subtract the value of those pennies from remaining cents
            cents = cents - 1;
        }
    }

    int total_coins = quarters + dimes + nickels + pennies;
    return total_coins;
}

File: problem_sets/week_06/sentimental-cash/cash.py
from cs50 import get_float

# Get non-negative value from user
while True:
    changes = get_float("Change: ")
    if changes >= 0:
        break


def cal_min_coins(n):
    # Track total coins
    coins = 0
    remain = int(n*100)

    while remain != 0:
        # quarters
        if remain >= 25:
            remain -= 25
            coins += 1
        # dimes
        elif remain >= 10:
            remain -= 10
            coins += 1
        # nickels
        elif remain >= 5:
            remain -= 5
            coins += 1
        # pennies
        else:
            remain -= 1
            coins += 1

    return coins


print(cal_min_coins(changes))
